drivers/RawSerial.h

```diff
@@ -71,31 +77,41 @@ class RawSerial: public SerialBase, private NonCopyable<RawSerial> {
      *  @note
      *    Either tx or rx may be specified as NC if unused
      */
+    MBED_DEPRECATED("The class has been deprecated and will be removed in the future.")
     RawSerial(PinName tx, PinName rx, int baud = MBED_CONF_PLATFORM_DEFAULT_SERIAL_BAUD_RATE);
 
-    /** Write a char to the serial port
+    /** @deprecated
+     * Write a char to the serial port
      *
      * @param c The char to write
      *
      * @returns The written char or -1 if an error occurred
      */
+    MBED_DEPRECATED("The class has been deprecated and will be removed in the future.")
```

Comment:
Do you need to put method deprecations in too? Not sure how well class deprecations work on all toolchains - if they don't you could just settle for deprecation on the constructors.

Reply:
It's not that important. No need to change this PR.

IMO, deprecating every method could lead to an awful lot of warnings.

I'm pretty certain deprecating constructors and class is sufficient - consider that in future. If only to minimise the work.